### PR TITLE
Add Dash Bootstrap Components

### PR DESCRIPTION
**Pull request by xhlulu**
Post-merge checklist:

The master branch is auto-deployed to `dash.plotly.com`.
Once you have merged your PR, wait 5-10 minutes and check dash.plotly.com
to verify that your changes have been made.

- [ ] I understand

If this PR documents a new feature of Dash:

- [ ] Comment on the original Dash issue with a link to the new docs.
- [ ] Reply to any community thread(s) asking for this feature.

If this PR includes a new dataset available at a remote URL:
- [ ] I have added this dataset to the `datasets/` folder
- [ ] I have added a mapping between the remote URL and the filename in the
`datasets/` folder into the `find_and_replace` dict in `dash_docs/tools.py`

If this PR adds an image or animated GIF:
- [ ] This image was saved and referenced locally rather than via an external link

If I introduced a new relative link inside `dcc.Markdown`:
- [ ] I considered whether I could replace the `dcc.Markdown` call with `rc.Markdown`, which will replace relative links with `tools.relpath` internally. Otherwise, I used e.g. `<dccLink href=tools.relpath('/layout') children="the first chapter"/>` instead of `[the first chapter](/layout)` (importing `tools` from `dash_docs`), and set `dangerously_allow_html=true` in the `dcc.Markdown` call.

If I changed the `chapter_index` by removing or relocating a page:
- [ ] I added a redirect in `dash_docs/server.py` from the old URL to the new URL

### CODE REVIEW

File: dash_docs/chapter_index.py [hunk 1]
```diff
@@ -633,6 +633,22 @@ def component_list(
...
+                'chapters': [
+                    {
+                        'url': 'https://dash-bootstrap-components.opensource.faculty.ai/',
+                        'description': (
+                            'A library of Bootstrap components '
+                            'created by [faculty.ai](https://faculty.ai/). '
+                            'Dash Boostrap Components makes it easier to '
+                            'build consistently styled '
+                            'apps with complex, responsive layouts.'
+                        )
```
Comment (rpkyle) [inline dash_docs/chapter_index.py:647]: @xhlulu Should we add this on the R side as well?

Comment (xhlulu) [reply dash_docs/chapter_index.py:647]: I'm unsure whether DBC supports R, they don't have any docs on R in their webapge: https://dash-bootstrap-components.opensource.faculty.ai/

**Comment by xhlulu** [comment]
Any idea why [the tests](https://app.circleci.com/pipelines/github/plotly/dash-docs/456/workflows/19e1fdf3-6750-45c6-8656-5888028cffb0/jobs/11938/tests) are failing? This is the output:
```
tests/integration/conftest.py:12: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
venv/src/dash/dash/testing/application_runners.py:76: in __call__
    return self.start(*args, **kwargs)
venv/src/dash/dash/testing/application_runners.py:209: in start
    wait.until(lambda: self.accessible(self.url), timeout=start_timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

wait_cond = <function ProcessRunner.start.<locals>.<lambda> at 0x7f790b59fdd0>
timeout = 60, poll = 0.1, msg = 'expected condition not met within timeout'

    def until(
        wait_cond, timeout, poll=0.1, msg="expected condition not met within timeout"
    ):  # noqa: C0330
        res = wait_cond()
        logger.debug(
            "start wait.until with method, timeout, poll => %s %s %s",
            wait_cond,
            timeout,
            poll,
        )
        end_time = time.time() + timeout
        while not res:
            if time.time() > end_time:
>               raise TestingTimeoutError(msg)
E               dash.testing.errors.TestingTimeoutError: expected condition not met within timeout

venv/src/dash/dash/testing/wait.py:25: TestingTimeoutError
```

I'm not totally sure how i broke the tests; looking at the heroku deployment, it seems fine.

**Comment by chriddyp** [comment]
@xhlulu - Could you try rerunning the tests? Could be an intermittent failure.

File: dash_docs/chapter_index.py (hunk 1 above)
Comment (rpkyle) [reply dash_docs/chapter_index.py:647]: @xhlulu Yes, it's supported now 🎉 

https://dash-bootstrap-components.opensource.faculty.ai/docs/dashr/

Comment (xhlulu) [reply dash_docs/chapter_index.py:647]: Awesome! In this case i think it's a good idea if they are added to the R side.

**Comment by alexcjohnson** [comment]
A few questions / issues:
- The sidebar link does nothing but open an empty menu - special handling of this case, to keep the link looking as it does but add a clickable entry inside it?
Screen Shot 2020-06-22 at 12 00 14 PM
- Do we want to do something to mark external links like this? Frustrating that this [isn't in unicode](https://www.quora.com/Is-the-symbol-for-external-link-available-in-Unicode-If-so-how-do-I-get-in-on-my-Mac) but we could add it as SVG.
- @xhlulu are you able to figure out adding the link in R or do you need @rpkyle to help?

**Comment by rpkyle** [comment]
> * @xhlulu are you able to figure out adding the link in R or do you need @rpkyle to help?

@xhlulu Let me know if there's anything you need from me, but it would be awesome to add in the DBC content for R here!

**Comment by xhlulu** [comment]
> > * @xhlulu are you able to figure out adding the link in R or do you need @rpkyle to help?
> 
> @xhlulu Let me know if there's anything you need from me, but it would be awesome to add in the DBC content for R here!

I'm so sorry, for some reason I didn't see this. I haven't touched dash-R docs befor e(or R in general), so I'd be happy if @rpkyle you would like to add it to Dash-R docs.

**Comment by xhlulu** [comment]
@alexcjohnson 

> * Do we want to do something to mark external links like this? Frustrating that this isn't in unicode but we could add it as SVG.
> * The sidebar link does nothing but open an empty menu - special handling of this case, to keep the link looking as it does but add a clickable entry inside it?

What's the best way I can resolve those problems in this PR?

**Comment by xhlulu** [comment]
Actually, looking at this line:

https://github.com/plotly/dash-docs/blob/6715265ed9e57b13f82b9f14c73812d80ece7275/dash_docs/chapter_index.py#L536-L548

Would we be able to simply do `'content': None` here without breaking the app?

**Comment by alexcjohnson** [comment]
> Do we want to do something to mark external links like this? Frustrating that this isn't in unicode but we could add it as SVG.

Don't worry about it in this PR - https://github.com/plotly/dash-docs/issues/911

> The sidebar link does nothing but open an empty menu - special handling of this case, to keep the link looking as it does but add a clickable entry inside it?

> Would we be able to simply do 'content': None here without breaking the app?

I'm not sure - maybe that would work, maybe it would work to flatten it so this one looks more like the entries in for example the callbacks section, or maybe neither of those would work and it just needs special handling in the sidebar creator.

> I'd be happy if @rpkyle you would like to add it to Dash-R docs.

Let's finish up the Python side first, but then Ryan can you add the R equivalent in this PR before we merge? Hopefully soon we'll be at a point where everyone feels comfortable adding Py and R content side-by-side but we may need one or two more examples of that first 🙏

**Comment by xhlulu** [comment]
> > Do we want to do something to mark external links like this? Frustrating that this isn't in unicode but we could add it as SVG.
> 
> Don't worry about it in this PR - #911
> 
> > The sidebar link does nothing but open an empty menu - special handling of this case, to keep the link looking as it does but add a clickable entry inside it?
> 
> > Would we be able to simply do 'content': None here without breaking the app?
> 
> I'm not sure - maybe that would work, maybe it would work to flatten it so this one looks more like the entries in for example the callbacks section, or maybe neither of those would work and it just needs special handling in the sidebar creator.
> 
> > I'd be happy if @rpkyle you would like to add it to Dash-R docs.
> 
> Let's finish up the Python side first, but then Ryan can you add the R equivalent in this PR before we merge? Hopefully soon we'll be at a point where everyone feels comfortable adding Py and R content side-by-side but we may need one or two more examples of that first

Sounds good. I'll try out what i mentioned tomorrow and we can see how it goes.

**Comment by xhlulu** [comment]
I just added a "name" to the dictionary, now it shows up in the sidebar as well as the main page. Does that fix it?

**Comment by xhlulu** [comment]
If the sidebar looks the way we want, @rpkyle would you like to take over this PR to add the R docs?

**Comment by HammadTheOne** [comment]
I've updated the Dash R docs to match the entry for the Python docs, and changed `External Link` to `faculty.ai` so people are aware of where they're headed if they select it from the sidebar. 

Closing this PR.

**Pull request merged**